### PR TITLE
WSAP-10-add-new-location

### PR DESCRIPTION
**Pull request by KatBre**

### CODE REVIEW

File: src/test/java/kb/weather/location/LocationServiceTests.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                .latitude(54.5)
+                .longitude(18.5)
+                .build();
+        when(locationRepository.save(any())).thenReturn(testLocation);
+
+        //when
+        Location location = locationService.createLocation("Gdynia", "pomeranian", "Poland", 54.5, 18.5);
+
+        //then
+        assertThat(location.getCity()).isEqualTo("Gdynia");
```
Comment (klezner) [inline src/test/java/kb/weather/location/LocationServiceTests.java]: As Michal said:
> you check the values created by you for testing purposes, it does not test the functionality of that class
> next level, lets try to use ArgumentCaptor - https://www.baeldung.com/mockito-argumentcaptor
> eg.

```
// when
locationService.createLocation("Gdansk", "Pomeranian", "Poland", 54.3, 18.6);
// then
verify(locationRepository).save(locationArgumentCaptor.capture());
assertThat(locationArgumentCaptor.getValue().getRegion()).isEqualTo("Pomeranian");
```

**Review by MichalKulygin — COMMENTED** [review]
Agree with klezner

File: src/test/java/kb/weather/location/LocationServiceTests.java (hunk 1 above)
Comment (KatBre) [reply src/test/java/kb/weather/location/LocationServiceTests.java]: fixed

File: src/test/java/kb/weather/location/LocationServiceTests.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                .longitude(18.5)
+                .build();
+        when(locationRepository.save(any())).thenReturn(testLocation);
+
+        //when
+        Location location = locationService.createLocation("Gdynia", " ", "Poland", 54.5, 18.5);
+
+        //then
+        assertThat(location.getCity()).isEqualTo("Gdynia");
+        assertThat(location.getRegion()).isEqualTo(" ");
```
Comment (klezner) [inline src/test/java/kb/weather/location/LocationServiceTests.java]: assertThat(location.getRegion()).isNull();

File: src/test/java/kb/weather/location/LocationServiceTests.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                .longitude(18.5)
+                .build();
+        when(locationRepository.save(any())).thenReturn(testLocation);
+
+        //when
+        Location location = locationService.createLocation("Gdynia", null, "Poland", 54.5, 18.5);
+
+        //then
+        assertThat(location.getCity()).isEqualTo("Gdynia");
+        assertThat(location.getRegion()).isEqualTo(null);
```
Comment (klezner) [inline src/test/java/kb/weather/location/LocationServiceTests.java]: assertThat(location.getRegion()).isNull();

File: src/test/java/kb/weather/location/LocationServiceTests.java (hunk 2 above)
Comment (KatBre) [reply src/test/java/kb/weather/location/LocationServiceTests.java]: done

File: src/test/java/kb/weather/location/LocationServiceTests.java (hunk 3 above)
Comment (KatBre) [reply src/test/java/kb/weather/location/LocationServiceTests.java]: done

**Pull request closed without merging**